plugins/processors/awsapplicationsignals/internal/resolver/attributesresolver.go

```diff
@@ -0,0 +1,162 @@
+var DefaultInheritedAttributes = map[string]string{
+	semconv1.AttributeDeploymentEnvironment: attr.AWSLocalEnvironment,
+	attr.ResourceDetectionASG:               common.AttributeEC2AutoScalingGroupName,
+	attr.ResourceDetectionHostId:            common.AttributeEC2InstanceId,
+	attr.ResourceDetectionHostName:          attr.ResourceDetectionHostName,
+}
+
+type subResolver interface {
+	Process(attributes, resourceAttributes pcommon.Map) error
+	Stop(ctx context.Context) error
+}
+
+type attributesResolver struct {
+	subResolvers []subResolver
+}
+
+// create a new attributes resolver
+func NewAttributesResolver(resolvers []appsignalsconfig.Resolver, logger *zap.Logger) *attributesResolver {
+	subResolvers := []subResolver{}
+	for _, resolver := range resolvers {
+		switch resolver.Platform {
+		case appsignalsconfig.PlatformEKS, appsignalsconfig.PlatformK8s:
+			subResolvers = append(subResolvers, getKubernetesResolver(resolver.Platform, resolver.Name, logger), newKubernetesResourceAttributesResolver(resolver.Platform, resolver.Name))
+		case appsignalsconfig.PlatformEC2:
+			subResolvers = append(subResolvers, newResourceAttributesResolver(resolver.Platform, AttributePlatformEC2, DefaultInheritedAttributes))
+		default:
+			if ecsutil.GetECSUtilSingleton().IsECS() {
+				subResolvers = append(subResolvers, newResourceAttributesResolver(appsignalsconfig.PlatformECS, AttributePlatformGeneric, DefaultInheritedAttributes))
+			} else {
+				subResolvers = append(subResolvers, newResourceAttributesResolver(resolver.Platform, AttributePlatformGeneric, GenericInheritedAttributes))
+			}
+		}
+	}
+	return &attributesResolver{
+		subResolvers: subResolvers,
+	}
+}
+
+// Process the attributes
+func (r *attributesResolver) Process(attributes, resourceAttributes pcommon.Map, _ bool) error {
+	for _, subResolver := range r.subResolvers {
+		if err := subResolver.Process(attributes, resourceAttributes); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *attributesResolver) Stop(ctx context.Context) error {
+	var errs error
+	for _, subResolver := range r.subResolvers {
+		if err := subResolver.Stop(ctx); err != nil {
+			errs = errors.Join(errs, err)
+		}
+	}
+	return errs
+}
+
+type resourceAttributesResolver struct {
+	defaultEnvPrefix string
+	platformType     string
+	attributeMap     map[string]string
+}
+
+func newResourceAttributesResolver(defaultEnvPrefix, platformType string, attributeMap map[string]string) *resourceAttributesResolver {
+	return &resourceAttributesResolver{
+		defaultEnvPrefix: defaultEnvPrefix,
+		platformType:     platformType,
+		attributeMap:     attributeMap,
+	}
+}
+func (h *resourceAttributesResolver) Process(attributes, resourceAttributes pcommon.Map) error {
+	for attrKey, mappingKey := range h.attributeMap {
+		if val, ok := resourceAttributes.Get(attrKey); ok {
+			attributes.PutStr(mappingKey, val.Str())
```

Comment:
So we're duplicating the value under a different name? Are we dropping the original key-value somewhere else?

Reply:
More importantly, we copy them from `resourceAttributes` to `attributes`.

Reply:
No, we don't drop the original ones from `resourceAttributes`.